rockcraft/oci.py

```diff
@@ -215,34 +219,86 @@ def to_oci_archive(self, tag: str, filename: str) -> None:
         src_path = self.path / f"{name}:{tag}"
         _copy_image(f"oci:{str(src_path)}", f"oci-archive:{filename}:{tag}")
 
-    def set_entrypoint(self, entrypoint: List[str]) -> None:
-        """Set the OCI image entrypoint.
-
-        :param entrypoint: The default list of arguments to use as the command to
-            execute when the container starts.
-        """
+    def set_entrypoint(self) -> None:
+        """Set the OCI image entrypoint. It is always Pebble and CMD is null."""
         emit.progress("Configuring entrypoint...")
         image_path = self.path / self.image_name
+        entrypoint = [f"/{self._PEBBLE_BINARY_PATH}", "enter"]
         params = ["--clear=config.entrypoint"]
         for entry in entrypoint:
             params.extend(["--config.entrypoint", entry])
         _config_image(image_path, params)
+        # Clear the CMD
+        _config_image(image_path, ["--clear=config.cmd"])
         emit.progress(f"Entrypoint set to {entrypoint}", permanent=True)
 
-    def set_cmd(self, cmd: List[str]) -> None:
-        """Set the OCI image default command parameters.
-
-        :param cmd: The default arguments to the entrypoint of the container. If
-            entrypoint is not defined, the first entry of cmd is the executable to
-            run when the container starts.
+    def set_pebble_services(
+        self,
+        services: Dict[str, Any],
+        name: str,
+        tag: str,
+        summary: str,
+        description: str,
+        base_layer_dir: Path,
```

Comment:
It is the base's filesystem, yes. 

Users could try to do that indeed. I propose we address that in a separate issue/PR, cause I believe that belongs to the broader story of "supporting user-defined Pebble". 

This is one of the items I have in the roadmap for us to discuss, so I'd rather push it for later instead of making a rush throw-away implementation right now. I've edited https://github.com/canonical/rockcraft/issues/65 to reference this discussion